### PR TITLE
MGMT-1072 assisted-insaller logs looks bad - ignore partial lines from coreos-installer output.

### PR DESCRIPTION
**Pull request by eranco74**

### CODE REVIEW

**Comment by eranco74** [comment]
Note the coreosInstallerLogger (will be renamed) can (and will) monitor the output and send host progress info for writing the image to disk.

**Comment by eranco74** [comment]
`
[root@test-infra-cluster-master-1 core]# tail -f /var/log/assisted-installer.log 
time="2020-06-11T14:58:44Z" level=info msg="Assisted installer started. Configuration is:\n {Role:master ClusterID:667268b2-845c-4a0a-95eb-39f5397c32c4 HostID:37077cba-72a8-4fb3-845e-5a431bac4941 Device:/dev/vda Host:10.19.176.63 Port:6000 Verbose:false OpenshiftVersion:4.5}"
time="2020-06-11T14:58:44Z" level=info msg="Installing node with role: master"
time="2020-06-11T14:58:44Z" level=info msg="Updating node installation status: Starting installation"
time="2020-06-11T14:58:44Z" level=info msg="Creating directory: /opt/install-dir"
time="2020-06-11T14:58:44Z" level=info msg="Updating node installation status: Installing as master"
time="2020-06-11T14:58:44Z" level=info msg="Getting master.ign file"
time="2020-06-11T14:58:44Z" level=info msg="Updating node installation status: Writing image to disk"
time="2020-06-11T14:58:44Z" level=info msg="Going to use image: https://mirror.openshift.com/pub/openshift-v4/dependencies/rhcos/4.5/latest/rhcos-4.5.0-0.nightly-2020-03-18-115438-x86_64-metal.x86_64.raw.gz"
time="2020-06-11T14:58:44Z" level=info msg="Writing image and ignition to disk"
time="2020-06-11T14:58:55Z" level=info msg="> Read disk 8.1 MiB/844.7 MiB (0%)   \r"
time="2020-06-11T14:58:56Z" level=info msg="> Read disk 9.2 MiB/844.7 MiB (1%)   \r"
time="2020-06-11T14:58:57Z" level=info msg="> Read disk 10.6 MiB/844.7 MiB (1%)   \r"
time="2020-06-11T14:58:58Z" level=info msg="> Read disk 12.3 MiB/844.7 MiB (1%)   \r"
time="2020-06-11T14:58:59Z" level=info msg="> Read disk 13.9 MiB/844.7 MiB (1%)   \r"

`

File: src/utils/utils.go [hunk 1]
```diff
@@ -41,6 +42,24 @@ func InitLogger(verbose bool) *logrus.Logger {
 	return log
 }
 
+type CoreosInstallerLogWriter struct {
+	progressRegex *regexp.Regexp
+	log           *logrus.Logger
+	lastLogLine   []byte
```
Comment (ronniel1) [inline src/utils/utils.go:48]: I don't see that this is used any where

File: src/utils/utils.go [hunk 2]
```diff
@@ -41,6 +42,24 @@ func InitLogger(verbose bool) *logrus.Logger {
...
+func NewCoreosInstallerLogWriter(logger *logrus.Logger) *CoreosInstallerLogWriter {
+	re := regexp.MustCompile(`^>(.*?)\((.*?)\)\s*\r`)
+	return &CoreosInstallerLogWriter{re, logger, []byte{}}
+}
+
+func (l *CoreosInstallerLogWriter) Write(p []byte) (n int, err error) {
+	if l.progressRegex.Match(p) {
+		l.log.Info(string(p))
+	}
+	return len(p), nil
```
Comment (ronniel1) [inline src/utils/utils.go:60]: Don't we need to add these bytes to the lastLogLine?

File: src/utils/utils.go (hunk 1 above)
Comment (eranco74) [reply src/utils/utils.go:48]: You are right, it's for the next task (reporting progress of the coreos-installer) I'll remove it for now.

File: src/utils/utils.go (hunk 2 above)
Comment (eranco74) [reply src/utils/utils.go:60]: I'll add it once it's required.

**Pull request merged**